src/main/generic/network/NetworkAgent.js

```diff
@@ -199,7 +201,8 @@ class NetworkAgent extends Observable {
             this._channel,
             msg.version,
             msg.startHeight,
-            msg.totalWork
+            msg.totalWork,
+            currentTimestampUTC - peerAddress.timestamp
```

Comment:
Thanks, that makes a lot of sense. I just updated the code to add the comment, please let me know if it works or if I can improve it further.